server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -481,7 +481,7 @@ public static MergeResult build(
             ObjectMapper existing,
             Mapper mergeWith,
             MergeReason reason,
-            MapperBuilderContext parentBuilderContext
+            MapperMergeContext parentMergeContext
```

Comment:
I was double checking the need for createChildContext in the new context, as opposed to calling it directly against the builder. Would that be an option? It looks like MergeResult#build could also take a MapperBuilderContext instead? Though I think there are still places where createChildContext is needed on the MapperMergeContext because we both build and merge mappers.

Reply:
>  It looks like MergeResult#build could also take a MapperBuilderContext instead?

Hm, I'm not certain if that's feasible. The reason is that `MergeResult#build` passes the `MapperMergeContext` to `MergeResult#buildMergedMappers`, which calls `objectMapper.merge(mergeWithMapper, reason, objectMergeContext)` and thus needs to have a `MapperMergeContext`. So it doesn't seem like a `MapperBuilderContext` would suffice.

Reply:
ok